src/typesystem.cpp

```diff
@@ -53,55 +53,57 @@ TypeSystemCore::TypeSystemCore() {
 		}
 
 	};
+
+	m_mono_core.unify_function = [&](int a, int b) {
```

Comment:
Maybe we should change this API so it takes a unifier by reference, instead of having to capture it?

Reply:
I did the first thing for it to work, so I'll take suggestions if you point me to an example to guide me.

Reply:
Have the std::function take a `unification::core&` as its first argument, then pass `*this` when you call it in `unify`

Reply:
Just an idea (which i do think is better), do whatever you think is better

Reply:
It seems unnecessary because I do need to capture `this` anyway, but it's clearer when you are using different cores so I'm leaving it